src/main/java/com/example/demo/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,86 @@
+import com.example.demo.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.demo.mapper.cartitem.CartItemMapper;
+import com.example.demo.mapper.shoppingcart.ShoppingCartMapper;
+import com.example.demo.model.CartItem;
+import com.example.demo.model.ShoppingCart;
+import com.example.demo.model.User;
+import com.example.demo.service.cartitem.CartItemService;
+import com.example.demo.service.shoppingcart.ShoppingCartService;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cart")
+@RequiredArgsConstructor
+public class ShoppingCartController {
+    private final ShoppingCartService shoppingCartService;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemService cartItemService;
+    private final CartItemMapper cartItemMapper;
+
+    @GetMapping
+    public ShoppingCartResponseDto getShoppingCart() {
+        Authentication authentication = SecurityContextHolder.getContext()
+                .getAuthentication();
+        String username = authentication.getName();
+        return shoppingCartService.getShoppingCartByUserEmail(username);
+    }
+
+    @PostMapping
+    public ShoppingCartResponseDto createShoppingCart(
+            @RequestBody ShoppingCartRequestDto shoppingCartRequestDto) {
+        Authentication authentication = SecurityContextHolder.getContext()
+                .getAuthentication();
+        User user = (User) authentication.getPrincipal();
+        CartItem cartItem = cartItemMapper.shopingCartRequestDtoToCartItem(shoppingCartRequestDto);
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setCartItems(Set.of(cartItem));
+        shoppingCart.setUser(user);
+        cartItem.setShoppingCart(shoppingCart);
+        ShoppingCartResponseDto shoppingCartResponseDto = shoppingCartService
+                .saveShoppingCart(shoppingCart);
+        return shoppingCartResponseDto;
+    }
+
+    @PutMapping("/items{id}")
+    public ShoppingCartResponseDto updateShoppingCart(@PathVariable Long id,
+                                                      @RequestBody
+                                                      ShoppingCartRequestDtoWithoutBookId
+                                                              shoppingCartRequestDtoWithoutBookId) {
```

Comment:
create and provide `UpdateCartItemDto` with `int quantity` instead of `ShoppingCartRequestDtoWithoutBookId`

```suggestion
    public ShoppingCartResponseDto updateShoppingCart(@PathVariable Long id,
                                                      @RequestBody @Valid 
                                                      UpdateCartItemDto
                                                              updateCartItemDto) {
```